src/compiler/checker.ts

```diff
@@ -3018,12 +3129,106 @@ namespace ts {
                     }
                 }
             }
+
+            function typeParametersToTypeParameterDeclarations(symbol: Symbol, context: NodeBuilderContext) {
+                let typeParameterNodes: ReadonlyArray<TypeParameterDeclaration> | undefined;
+                const targetSymbol = getTargetSymbol(symbol);
+                if (targetSymbol.flags & (SymbolFlags.Class | SymbolFlags.Interface | SymbolFlags.TypeAlias)) {
+                    typeParameterNodes = map(getLocalTypeParametersOfClassOrInterfaceOrTypeAlias(symbol), tp => typeParameterToDeclaration(tp, context));

```

Comment:
why is the new map over typeParameterToDeclaration needed here, and why is the map over the symbol's mapper needed in the instantiated case?

Reply:
I think it has something to do with the switch to parentSymbol/nextSymbol instead of the weird parentSymbol-only code before.

Reply:
So before `parentSymbol` was the symbol left of the current symbol in the chain - now `parentSymbol` is the current symbol, and `nextSymbol` is the one to the right of the current one. We say "if the symbol on the right is instantiated, then it must have come from an instantiated version of the symbol on the left - get its mapper and apply it to the type parameters of the symbol on the left". It was the same in the symbol display builder, except it was more straightforwardly written, since it built the output string from left to right, whereas we build the node tree from right to left.